### PR TITLE
[dynamo][refactor] Move some helper functions to global scope

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #120520
* __->__ #120426

This is to prepare for guard C++ refactor work.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: torch/_dynamo/guards.py [hunk 1]
```diff
@@ -169,6 +169,45 @@ def strip_getattr_getitem(name):
...
+        torch._C._dispatch_keys(value) | torch._C._dispatch_tls_local_include_set()
+    ) - torch._C._dispatch_tls_local_exclude_set()
+    dtype = value.dtype
+    device_index = value.device.index
+    requires_grad = value.requires_grad
+    guard_str = (
+        f"check_tensor({name}, {pytype.__qualname__}, {dispatch_key}, {dtype}, "
+        f"device={device_index}, requires_grad={requires_grad}, size={sizes}, stride={strides})"
+    )
+    return guard_str
```
Comment (ezyang) [inline torch/_dynamo/guards.py:208]: Consider adding docs / type signatures!

**Review by ezyang — APPROVED** [review]
okey dokey

File: torch/_dynamo/guards.py (hunk 1 above)
Comment (anijain2305) [reply torch/_dynamo/guards.py:208]: Will do that in the followup PRs.

**Pull request closed without merging**